### PR TITLE
votekick print to party chat.

### PR DESCRIPTION
**Pull request by surepy**
> experimental: [cat] votekick called: Jerry Seinfeld => Mr. Lean 2.O (#TF_vote_kick_player_other)

> sulfix: kick called by Jerry Seinfeld

also fixed possible crash on killsay.

### CODE REVIEW

**Comment by surepy** [comment]
wait- it fetches the wrong person. will commit again after fixing

**Comment by surepy** [comment]
it should prob work now.

File: src/votelogger.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 static settings::Bool requeue{ "hack.requeue-on-kick", "false" };
 static settings::Bool vote_kicky{ "hack.autovote.yes", "false" };
 static settings::Bool vote_kickn{ "hack.autovote.no", "false" };
+static settings::Bool party_say{ "hack.partysay", "true" };
```
Comment (TotallyNotElite) [inline src/votelogger.cpp]: We really have to be more creative with rvar names.

File: src/votelogger.cpp [hunk 2]
```diff
@@ -58,6 +58,17 @@ void dispatchUserMessage(bf_read &buffer, int type)
                 playerlist::AccessData(info.friendsID).state ==
                     playerlist::k_EState::DEFAULT)
                 g_IEngine->ClientCmd_Unrestricted("vote option1");
+        if (*party_say)
+        {
+        	g_IEngine->GetPlayerInfo(caller, &info2);
```
Comment (TotallyNotElite) [inline src/votelogger.cpp]: This function returns a bool that should be checked for.

File: src/votelogger.cpp [hunk 3]
```diff
@@ -58,6 +58,17 @@ void dispatchUserMessage(bf_read &buffer, int type)
...
+        	g_IEngine->ExecuteClientCmd(format("say_party [cat] votekick called: ",
+        			boost::replace_all_copy((std::string)info2.name, ";", ";"),
```
Comment (TotallyNotElite) [inline src/votelogger.cpp]: Is a cast to std::string really necessary? IIRC, implicit conversions from c_str to std::string exist.

File: src/votelogger.cpp [hunk 4]
```diff
@@ -58,6 +58,17 @@ void dispatchUserMessage(bf_read &buffer, int type)
...
+					" => ",
+        			boost::replace_all_copy((std::string)info.name, ";", ";"),
+        			" (",reason,")").c_str());
+        	//	because tf2 is stupid and doesn't have escape characters, i'll use the greek question mark instead.
```
Comment (TotallyNotElite) [inline src/votelogger.cpp]: These comments should be placed above the relevant code.

File: src/votelogger.cpp (hunk 3 above)
Comment (BenCat07) [reply src/votelogger.cpp]: Also PLS [CAT] in uppercase

File: src/votelogger.cpp [hunk 5]
```diff
@@ -41,7 +41,7 @@ void dispatchUserMessage(bf_read &buffer, int type)
         eid >>= 1;
 
         unsigned steamID = 0;
-        player_info_s info{};
+        player_info_s info{}, info2{};	//	info is the person getting kicked, info2 is the person calling the kick.;
```
Comment (BenCat07) [inline src/votelogger.cpp]: Comments. above. the. code

File: src/votelogger.cpp [hunk 6]
```diff
@@ -29,9 +30,8 @@ void dispatchUserMessage(bf_read &buffer, int type)
     {
         // TODO: Add always vote no/vote no on friends. Cvar is "vote option2"
         was_local_player = false;
-        auto caller      = (unsigned char) buffer.ReadByte();
-        // unknown
-        buffer.ReadByte();
+        int team	=	buffer.ReadByte();
+        int caller      =	buffer.ReadByte();	//	client id of init apparently
```
Comment (BenCat07) [inline src/votelogger.cpp]: Comments.above.the.code

File: src/hacks/KillSay.cpp [hunk 7]
```diff
@@ -58,11 +58,15 @@ std::string ComposeKillSay(IGameEvent *event)
     if (g_IEngine->GetPlayerForUserID(kid) != g_IEngine->GetLocalPlayer())
         return "";
     std::string msg = source->at(rand() % source->size());
-    while (msg == lastmsg)
+    if(source->size() != 1)
     {
-        msg = source->at(rand() % source->size());
+    	//	checks if the killsays.txt file is not 1 line. 100% sure it's going to crash if it is.
+    	while (msg == lastmsg)
+    	{
```
Comment (BenCat07) [inline src/hacks/KillSay.cpp]: nice {} u sell

File: src/votelogger.cpp [hunk 8]
```diff
@@ -58,14 +61,32 @@ void dispatchUserMessage(bf_read &buffer, int type)
...
                 g_IEngine->ClientCmd_Unrestricted("vote option1");
+        if (*party_say)
+        {
+            if (g_IEngine->GetPlayerInfo(caller, &info2))
+            {
+                // because tf2 is stupid and doesn't have escape characters,
+                // use the greek question marks instead. big brain.
+                // Clang-format why, TODO: Don't use format func
+                g_IEngine->ExecuteClientCmd(
+                    format("say_party [cat] votekick called: ",
```
Comment (BenCat07) [inline src/votelogger.cpp]: [CAT] not [cat]

File: src/hacks/KillSay.cpp (hunk 7 above)
Comment (TotallyNotElite) [reply src/hacks/KillSay.cpp]: How about no.

**Pull request merged**